Update edxapp default image to hawthorn.1-2.0.1

sampaccoud opened the pull request:
## Purpose

This image is a complete refactorization of the build process and
fixes issues on settings that were blocking some deployments like
mongodb port numbers not being formatted to integers.


## Proposal

Change image tag in variables.

:warning: depends on #137 :warning:

sampaccoud committed:
⬆️(edxapp) update edxapp default image to hawthorn.1-2.0.0

This image is a complete refactorization of the build process and
fixes issues on settings that were blocking some deployments like
mongodb port numbers not being formatted to integers. The
webpack-stats.json is now already copied to a location collected
by collectstatic so we should not try to copy it anymore in Arnold.

jmaupetit committed:
🐛(edxapp) use requests to download fixtures

We no longer install curl in edxapp production image. We decided to use
raw python & requests to download fixtures instead of adding a new
dependency.

jmaupetit committed:
🐛(edxapp) fix demo course release

When upgrading to hawthorn, we've missed the demo course release
upgrade. This is now fixed.

jmaupetit committed:
🐛(edxapp) add npm to the LMS build configuration

We need to run the paver update_assets command to compile a custom LMS
theme. It appears that even if we ask paver not to install
prerequisites, it requires npm to be installed... ¯\_(ツ)_/¯

Thus npm is now installed in the LMS buildConfiguration before compiling
the theme.

sampaccoud commented on apps/edxapp/templates/lms/bc.yml.j2:
```diff
@@ -39,8 +39,12 @@ spec:
+        rm -rf /var/lib/apt/lists/*
       COPY . /edx/app/edxapp/edx-platform/themes/custom-theme
-      RUN NO_PREREQ_INSTALL=1 paver update_assets --settings={{ edxapp_build_settings }} --skip-collect
+      RUN PATH=${PWD}/node_modules/.bin:${PATH} NO_PREREQ_INSTALL=1 paver update_assets --settings={{ edxapp_build_settings }} --skip-collect
```
@jmaupetit  this can be removed now?

jmaupetit replied:
Yes, once we've upgraded to `hawthorn.1-2.0.1`

sampaccoud commented on apps/edxapp/vars/all/main.yml:
```diff
@@ -6,7 +6,7 @@ edxapp_lms_host: "lms.{{ project_name}}.{{ domain_name }}"
 # -- edxapp (cms/lms)
 edxapp_image_name: "fundocker/edxapp"
-edxapp_image_tag: "hawthorn.1-1.0.6"
+edxapp_image_tag: "hawthorn.1-2.0.0"
```
upgrade to 2.0.1?

jmaupetit replied:
Yes, we need to make a new edxapp release, upgrade to 2.0.1 and remove the PATH (your previous comment).

jmaupetit committed:
🐛(edxapp) add node binaries to the PATH

Switch to hawthorn.1-2.0.1 that adds node binaries to the PATH so that
we are not constraint to update the PATH everytime we need to access to
a node tool.

jmaupetit committed:
✨(acme) generate live ssl certificate for preprod

As the preproduction environment will be showned to customers, it would
be more comfortable for them to avoid having to accept insecure
certificates.

sampaccoud commented:
I can't approve it because I opened it but it definitely looks GTM!

jmaupetit commented:
It's odd (strange) but I can :wink:
